### ISSUE TITLE
'this' in method is undefined?

### ISSUE THREAD

**Issue by wayshall** [babel/babel#3035]
hi all!
I use bable to translate es6 in a angularjs project.

angularjs: 1.x 
babel: 6.1.4

my code is below:
<pre><code>
angular.module('app.services', [])
.service('appService', AppService);

class AppService {
    /*@ngInject*/
    constructor(http){
        this.http = http;
    }
    getList(page){
        console.log('getList this....')
        console.dir(this)
        return this.http.getPage('/propertyController.do?datagrid', page);
    }
}
</code>
</pre>
the console in the chrome show the 'this' in the 'getList' method is undefined!
It's a very simple demo, why is wrong.....

**Comment by wayshall** [comment]
I check and found the generated code by babel is not correct.
I try to add and modify some code and it run normal....
![es6](https://cloud.githubusercontent.com/assets/965315/11143492/8f7084ca-8a30-11e5-9542-5f749685e7c7.png)

**Comment by weikinhuang** [comment]
I am also encountering this issue

```json
// packages
    "babel-core": "6.1.20",
    "babel-plugin-transform-es3-member-expression-literals": "6.1.18",
    "babel-plugin-transform-es3-property-literals": "6.1.18",
    "babel-plugin-transform-es5-property-mutators": "6.1.18",
    "babel-plugin-transform-node-env-inline": "6.1.18",
    "babel-plugin-transform-object-assign": "6.1.18",
    "babel-polyfill": "6.1.19",
    "babel-preset-es2015": "6.1.18",
    "babel-preset-stage-1": "6.1.18",

// config
  "browserify": {
    "transform": [
      [
        "babelify",
        {
          "sourceMapRelative": "./public/static",
          "presets": [
            "es2015",
            "stage-1"
          ],
          "plugins": [
            "transform-es3-member-expression-literals",
            "transform-es3-property-literals",
            "transform-es5-property-mutators",
            "transform-node-env-inline",
            "transform-object-assign"
          ]
        }
      ],
      "brfs"
    ]
  },
```

```javascript
export default class IndexViewCtrl {
	constructor(UserSessionSvc, FLAGS) {
		this.FLAGS = FLAGS;
		this.appTitle = FLAGS.title;
		this.isLoggedIn = false;

		UserSessionSvc.getLoggedInUser()
			.then(() => this.isLoggedIn = true);
	}
}

IndexViewCtrl.$inject = [
	"UserSessionSvc",
	"FLAGS"
];
```

becomes
```javascript
var IndexViewCtrl = function IndexViewCtrl(UserSessionSvc, FLAGS) {
	_classCallCheck(this, IndexViewCtrl);

	this.FLAGS = FLAGS;
	this.appTitle = FLAGS.title;
	this.isLoggedIn = false;

	UserSessionSvc.getLoggedInUser().then(function () {
		return _this.isLoggedIn = true;
	});
};

exports.default = IndexViewCtrl;

IndexViewCtrl.$inject = ["UserSessionSvc", "FLAGS"];
```

`_this` is never defined in the constructor definition of the generated code.

**Comment by thejameskyle** [comment]
I'm not able to reproduce either scenario. @weikinhuang I used yours because it included the presets/plugins here:

https://github.com/thejameskyle/__babel-bug-3035

Both of you should try updating your dependencies and see if it's fixed.

I'm going to close this, if someone can fork that repo above and reproduce the issue I'll reopen this.

**Comment by weikinhuang** [comment]
@thejameskyle I am also unable to reproduce it in your repo. But I am inconsistently encountering this issue in some of the defined classes, not all of them. I am using the latest versions of babelify and browserify and I have no idea why it's only happening to some classes and not others.

**Comment by wayshall** [comment]
hi @thejameskyle !
I had been fork your repo, and modify some code to reproduce this issue for my case.
you can check this index.js file to confirm: https://github.com/wayshall/__babel-bug-3035/blob/master/index.js .

'this' will be print undefined when the object's method of class send to another object's method and invoke in it.

**Comment by weikinhuang** [comment]
@thejameskyle I managed to reproduce the bug by distilling down my repo to a few files. I forked your repo and opened a PR. My fork/branch https://github.com/weikinhuang/__babel-bug-3035/tree/bug It seems to broken in this EXACT configuration. If I move `index.js` up to the top level it works fine. If I get rid of the `babel` block in my `package.json` it also works fine. I think `babelify` is reading the babel configuration twice and therefore running through the presets process twice. I included the generated code from my gulp build as well.
